lib/tmdb/client.rb

```diff
@@ -201,20 +233,50 @@ def get_parsed_tv_episode_data(series_id:, season_number:, episode_number:)
         JSON.parse(open(url).read, symbolize_names: true)
       end
 
-      def get_parsed_tv_search_results(query)
-        url = "#{BASE_URL}/search/tv?api_key=#{API_KEY}&query=#{query}"
+      def get_parsed_tv_search_results(series_name)
+        searchable_name = I18n.transliterate(series_name)
+        url = "#{BASE_URL}/search/tv?api_key=#{API_KEY}&query=#{searchable_name}"
         JSON.parse(open(url).read, symbolize_names: true)&.dig(:results)
       end
 
       def get_parsed_multi_search_results(query)
-        url = "#{BASE_URL}/search/multi?api_key=#{API_KEY}&query=#{query}"
+        searchable_query = I18n.transliterate(query)
+        url = "#{BASE_URL}/search/multi?api_key=#{API_KEY}&query=#{searchable_query}"
         JSON.parse(open(url).read, symbolize_names: true)&.dig(:results)
       end
 
-      def get_parsed_movie_search_results(query)
-        url = "#{BASE_URL}/search/movie?api_key=#{API_KEY}&query=#{query}"
+      def url_for_person_search(person_name)
+        searchable_name = I18n.transliterate(person_name)
+        "#{BASE_URL}/search/person?api_key=#{API_KEY}&query=#{searchable_name}"
+      end
+
+      def get_parsed_movie_search_results(movie_title)
+        searchable_title = I18n.transliterate(movie_title)
+        url = "#{BASE_URL}/search/movie?api_key=#{API_KEY}&query=#{searchable_title}"
         JSON.parse(open(url).read, symbolize_names: true)&.dig(:results)
       end
+
+      def url_for_movie_discover_search(params)
+        url = "#{BASE_URL}/discover/movie?api_key=#{ENV['tmdb_api_key']}&certification_country=US"
+        url += "&with_people=#{params[:people]}" if params[:people].present?
+        url += "&with_genres=#{params[:genre]}" if params[:genre].present?
+        url += "&with_companies=#{params[:company]}" if params[:company].present?
+        url += "&certification=#{params[:mpaa_rating]}" if params[:mpaa_rating].present?
+        url += "&sort_by=#{params[:sort_by]}.desc" if params[:sort_by].present?
+        url += "&page=#{params[:page]}"
+        if params[:year].present? && params[:year_select].present?
+          url += "&primary_release_year=#{params[:year]}" if params[:year_select] == 'exact'
+          url += "&primary_release_date.lte=#{params[:year]}-01-01" if params[:year_select] == 'before'
+          url += "&primary_release_date.gte=#{params[:year]}-12-31" if params[:year_select] == 'after'
+        elsif params[:year].present?
+          url += "&primary_release_year=#{params[:year]}"
+        end
+        url
+      end
+
+      def get_data(url)
```

Comment:
🤔 I'm wondering about a clearer name here to indicate it's doing the actual api request.

Reply:
I can see what you mean. 🤔 I think we're exploring that totally different approach like `request(:person_search, params)` in PR #289. I'm hoping to chip away at this `tmdb_handler_discover_search` while making minimal changes to the other methods, then we can go back and replace all the api calls with that `request` method.

Reply:
Done.